manifests/params.pp

```diff
@@ -181,6 +185,7 @@
   $puppet_url                 = "https://${::fqdn}:8140"
   $puppet_use_environment_api = undef
   $puppet_use_cache           = undef
+  $puppet_envdir              = "${puppet_codedir}/environments"
```

Comment:
It's not DRY, but I don't see another way without introducing dependencies. I liked your fact solution so if that's merged into stdlib we should move to it instead.